ADR/0009-pipeline-service-via-operator.md

```diff
@@ -0,0 +1,53 @@
+# 0009 Pipeline Service via Operator
+
+Date: 2022-12-01
+
+## Status
+
+Proposed
+
+## Context
+
+kcp is no longer being used as a control plane for StoneSoup. This means that
+"Pipeline Service" cannot be deployed as an independent service. For our
+initial MVP, all Tekton APIs need to be deployed onto a standard OpenShift
+cluster (specifically OpenShift Dedicated).
+
+## Decision
+
+All Tekton APIs will be provided using the stock OpenShift Pipelines operator.
+In the spirit of developing in a "service first" manner, StoneSoup will deploy
+a candidate "nightly" release of the operator. The service will be defined in
+the [pipeline-service](https://github.com/openshift/pipeline-service)
+repository, which is then imported into
+[infra-deployments](https://github.com/redhat-appstudio/infra-deployments) as
+an ArgoCD application.
+
+Configurations that are specific to StoneSoup must be made available through
+the Pipelines operator CRDs. The following changes are specific to StoneSoup:
+
+- Disable Tekton Triggers. Pipelines as Code will be the favored mechanism for
+  event-based triggering of pipelines for now. This decision can be revisited
+  in the future based on need.
+- Disable the pruner that comes with the operator. Tekton Results will be used
```

Comment:
Do we consider this as "temporary" — because the pruner doesn't fit the bill today — or a finite decision ?
Asking because the requirements that Pipeline Service / StoneSoup has/will have are most likely also going to be very useful for our customers.

Reply:
Agree here - at the current juncture we are still learning, so we need maximum flexibility with respect to how we configure things for the service. Once we land on something that we believe works at scale, we can advocate for those pieces to be default/primary experiences in the operator.